src/lib/cfa/ntp_conf.rb

```diff
@@ -125,10 +125,17 @@ def each(&block)
         record_entries.each(&block)
       end
 
+      # Get last record in collection
+      def last
+        record_entries.last
+      end
+
       # Adds a new Record object to the collection.
+      # @note it creates new instance of record after this.
```

Comment:
it means, that you have instance of Record `r` and you call `records << r` but in fact in records, there is another instance and not `r` itself. So if you modify it later, like it happen in code, it is ignored.